Add Decky Cloud Save

GedasFX opened the pull request:
# Decky Cloud Save

A plugin based on [rclone](https://rclone.org/), which allows users to back-up game saves to the cloud. Supports [OneDrive](https://onedrive.live.com/), [Google Drive](https://drive.google.com/), [Dropbox](https://www.dropbox.com/), and [many more](#backend-support)!

#### Features

* Ability to sync game saves or arbitrary files to the cloud.
* A high variety of supported cloud providers, courtesy of [rclone](https://rclone.org/).
* Ability to back up files automatically after a game is closed.
* File counter, which estimates the number of files a sync path would pick up. Prevents accidental backing up of the entire steam deck.


IMPORTANT! This plugin does not support bidirectional sync. In other words, this plugin is not intended for use cases to sync saves between devices, but rather just to keep your game progress safe in case of data loss.


## Checklist:

### Developer Checklist

- [x] I am the original author or an authorized maintainer of this plugin.
- [x] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin Checklist

- [x] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [x] I have verified my plugin is unique or alternatively provides more/alternative functionality to a similar plugin already on the store.

<!-- The following section needs to be modified as yes/no answers by the plugin developer. -->

<!-- Ex: "**Yes/No**: ..." becomes "**Yes**: ..." -->

### Plugin Backend Checklist

- **No**: I am using a custom backend other than Python.
- **No**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library).
- **Yes**: I am using a custom binary that has all of it's dependencies statically linked.

<!-- The following section is for testers and maintainers, please do not modify it unless you don't meet the criteria for testing on the preview channel. -->

## Testing

- [x] Tested on SteamOS Stable Update Channel.
<!-- Leave me be unless you don't need preview testing, I just make the buttons look nice -->
- [x] Tested on SteamOS Beta Update Channel.
<!-- If you answered "No" to all of the Plugin Backend Checklist then you may remove the below line for testing on preview channel. -->
- [x] Tested on SteamOS Preview Update Channel.

AAGaming00 reviewed (approved):
code lgtm

GedasFX commented:
Hey, is there anything that I need to do to get this forward? I've been using this for over a month and works fine :D

TrainDoctor commented:
Please update DFL to latest version and fix any issues related to ModalRoot or NavigateToExternalWeb (if relevant). Then rebase and push to the PR. Thanks.

DJScias commented:
**Decky:** 2.5.4-pre1
**SteamOS:** 3.4.4 Stable and Beta

A) One major issue (two if we count Google Drive broken, but Google does not count Steam Browser as safe). Beta suffers from the "Cannot read property 'bind' issue. **Fix for this is updating DFL to latest.**
B) Single minor issues noticed so far, once you've linked something and press B to go back it'll instead go back to the browser that you made the linking steps to (like dropbox). Not sure if it's possible, but perhaps have B just mean go out of the "Configure Cloud Storage Provider" part?
C) Nay for production for now until DFL is updated.

GedasFX commented:
A) One major issue (two if we count Google Drive broken, but Google does not count Steam Browser as safe). Beta suffers from the "Cannot read property 'bind' issue. Fix for this is updating DFL to latest.
> Updated and retested. Works on Stable and Beta.
> For the Google, I removed that button entirely as it's not working. I can re-add it eventually (as I would need to remove the text anyway).

B) Single minor issues noticed so far, once you've linked something and press B to go back it'll instead go back to the browser that you made the linking steps to (like dropbox). Not sure if it's possible, but perhaps have B just mean go out of the "Configure Cloud Storage Provider" part?
> Fixed. Thank you @AAGaming00

GedasFX commented:
Also updated description to include the message.

TrainDoctor commented:
Plugin will need a DFL bump to fix recent issues, then once this bump is made it will be good for testing once more.

GedasFX commented:
@TrainDoctor you mean another bump?

GedasFX commented:
https://github.com/GedasFX/decky-cloud-save/commit/4184941f35fea5d8142887293fb5a2b1d303dc8b
I hope this was enough

GedasFX commented:
Added new features:
1. Added install scripts from the desktop mode. This readds the 'native' Google Drive support, as much as I physically can. https://github.com/GedasFX/decky-cloud-save/pull/4
2. Fixed bug where data appeared to be missing when things stopped loading (explained here https://discord.com/channels/960281551428522045/960284327445418044/1083076585541882016) https://github.com/GedasFX/decky-cloud-save/pull/5

Fisch03 commented:
Works on stable, beta and preview for me, i only tested with onedrive though.
File picker wouldn't load right after installing but restarting steam fixed that.
